jax/interpreters/xla.py

```diff
@@ -1474,7 +1500,7 @@ def _named_call_translation_rule(c, axis_env, in_nodes, name_stack, *,
   subc = xb.make_computation_builder(name)
   args = [xb.parameter(subc, i, c.GetShape(n)) for i, n in enumerate(in_nodes)]
   out_nodes = jaxpr_subcomp(subc, call_jaxpr, backend, axis_env, (),
-                            extend_name_stack(name_stack, name), *args)
+                            name_stack.extend(name), *args)
```

Comment:
It seems that this is the only place where we pass a string instead of a primitive. Even here I would pass the named_call primitive, if we want the stack to represent the stack of primitives.